sources/osgShader/Compiler.js

```diff
@@ -771,8 +771,19 @@ Compiler.prototype = {
             float: shadowedOutput
         } ).setShadowAttribute( shadow );
 
+        // allow overwrite by inheriting compiler
+        // where shadow inputs ( NDotL notably)
+        // can be used for non standard shadows
+        return this.doShadowLight( lightedOutput, shadowedOutput, shadowInputs );
+
+    },
+
+    doShadowLight: function ( lightedOutput, shadowedOutput ) {
```

Comment:
the name of the function is a bit generic, why not connectShadowLightNode or something that tell more what is it if you have better. I am not telling my suggestion is the best